src/torchpme/utils/tuning/grid_search.py

```diff
@@ -0,0 +1,212 @@
+#!/usr/bin/env python3
+
+import math
+import time
+from typing import Optional
+from warnings import warn
+
+import torch
+import vesin.torch
+
+from torchpme import (
+    CoulombPotential,
+    EwaldCalculator,
+    P3MCalculator,
+    PMECalculator,
+)
+from torchpme.utils import EwaldErrorBounds, P3MErrorBounds, PMEErrorBounds
+
+from . import _estimate_smearing_cutoff, _validate_parameters
+
+
+def grid_search(
+    method: str,
+    charges: torch.Tensor,
```

Comment:
I would turn the logic around and keep the `tune_XXX` method. Also, `grid_search` is a very common name. It is not really clear from this that this will find the optimal parameters for the methods.